pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -219,14 +221,30 @@ func (p *DefaultProvider) createLaunchTemplate(ctx context.Context, options *ami
 	if err != nil {
 		return ec2types.LaunchTemplate{}, err
 	}
+	createLaunchTemplateInput := p.GetCreateLaunchTemplateInput(options, p.ClusterIPFamily, userData)
+	output, err := p.ec2api.CreateLaunchTemplate(ctx, createLaunchTemplateInput)
+	if err != nil {
+		return ec2types.LaunchTemplate{}, err
+	}
+	log.FromContext(ctx).WithValues("id", aws.ToString(output.LaunchTemplate.LaunchTemplateId)).V(1).Info("created launch template")
+	return lo.FromPtr(output.LaunchTemplate), nil
+}
+
+// you need UserData, AmiID, tags, blockdevicemappings, instance profile,
+func (p *DefaultProvider) GetCreateLaunchTemplateInput(options *amifamily.LaunchTemplate, ClusterIPFamily corev1.IPFamily, userData string, dryrun ...bool) *ec2.CreateLaunchTemplateInput {
```

Comment:
Why do we need to add the `dryrun` field in side this `GetCreateLaunchTemplateInput`? What do you think about just adding the dryrun field once you get the values have the create launch template input in the validation controller